### ISSUE TITLE
Manager for proxied apps and ability to configure proxied apps CLI args via JupyterLab UI interface

### ISSUE THREAD

**Issue by mahendrapaipuri** [jupyterhub/jupyter-server-proxy#383]
### Proposed change

We use jupyter-server-proxy to give access to applications like tensorboard, mlflow, _etc_ to our users _via_ our JupyterHub deployment. These applications (certainly many more) needs CLI args (for instance `logdir` for tensorboard) to be able to work well. This can be tackled by using custom env vars or config files that users need to configure before launching the app. But it would be better if users can set these CLI args directly from the JupyterLab UI before launching the app. 

When user clicks the proxy app on the JupyterLab launcher, it shows an input dialog as below
![labappargs](https://user-images.githubusercontent.com/44365948/228313227-d83f54fb-fc5b-4582-8713-8431899a0ce3.png)
and then user can simply setup CLI args and/or env vars needed to launch that app. We will send these args to the backend, sanitize them and append to the proxy app command. 

Another feature is the ability to launch multiple instances of a given app. This can be achieved by adding one more [config](https://jupyter-server-proxy.readthedocs.io/en/latest/server-process.html#specifying-config-from-python-packages) option, say `num_instances`, and then registering the handlers with suffix `app_{instance_num}`. This way user can have multiple instances of a given app and it is admins who decide how many instances to allow.

With current implementation of jupyter server proxy, there is no way to terminate the app _via_ UI (user can always send a SIGTERM to process pid "manually"). The idea is to add a manager to the proxied apps that can be managed as kernels, terminals in the JupyterLab sidebar as below
![labapprunning](https://user-images.githubusercontent.com/44365948/228314303-8d98ee53-c38a-44bd-a4f2-9d34bd47c26b.png)
The users can terminate apps here, which will send a SIGTERM signal to the PID that we got from simpervisor.

### Who would use this feature?

All the users that use jupyter server proxy to get UI access to the applications running on a platform. This will be very useful particularly for HPC platforms where JupyterHub/JupyterLab is most common way to provide web access on the compute nodes.

### (Optional): Suggest a solution

We have already implemented these for our platform and it has running well for almost a year. If the developers of jupyter server proxy are interested, we would be happy to open a PR.

**Comment by ryanlovett** [comment]
@mahendrapaipuri Those seem like nice features to me! I would suggest creating separate PRs for each of the new features for discussion.

Would the parameterisation UI eventually account for all [server process options](https://jupyter-server-proxy.readthedocs.io/en/latest/server-process.html#server-process-options)?

The manager UI could be a good place to show available apps (i.e. all entry points) and the attributes of currently running servers, like the [assigned port numbers](https://github.com/jupyterhub/jupyter-server-proxy/issues/343#issuecomment-1440481262).

**Comment by mahendrapaipuri** [comment]
@ryanlovett Yes, the idea is to split it in different PRs to have cleaner discussion and review.

> Would the parameterisation UI eventually account for all [server process options](https://jupyter-server-proxy.readthedocs.io/en/latest/server-process.html#server-process-options)?

So, basically in my proposal we extend the callable arguments with CLI args and environment variables. That means the besides `port` and `base_url` in callable function, we have now `args` and `env` as well. To parameterise options like `map_path`, I think we will have to modify what we did for our deployment. Nevertheless I think it is a good idea to be able to parameterise other options as well and we already had few cases where this would be useful. I will look into it. Thanks!!

**Comment by bollwyvl** [comment]
#425 is in, which extends the labextension to work in the `/tree` view provided by Notebook 7.

The first pass tried to emulate the Classic behavior as closely as possible, with just a menu that always opens a new fat browser tab.

Revisiting this design: the block-the-world-modal would _still work_ in Notebook 7 (and has precedent for e.g. keyboard shortcuts) but some other extensions like _Running_ or the _Settings Editor_ are in tabs in the sidebar-less dockpanel in the 970px main area. 

Some treatment ideas:
- keep the menu
  - add an icon indicator for configurable proxies, e.g. :gear: or :wrench: which opened a new panel.
    - pop a new tab, labeled with the _Title_ of just the form
    - dispose of the tab when the proxy is launched
 - always open a new tab, e.g. _Proxies_
   - add some other treatment

**Comment by mahendrapaipuri** [comment]
@bollwyvl Cheers for the ideas. Before we can implement the configuration of proxy apps at runtime _via_ form, we need to finish working on #395 and merge it to `main`. I have worked on the remaining little issues and please have a look at it. I have added both [unit tests](https://github.com/jupyterhub/jupyter-server-proxy/blob/8987c544a61072db34f922d5ce03062302803ea5/tests/test_api.py) and [acceptance tests](https://github.com/jupyterhub/jupyter-server-proxy/blob/8987c544a61072db34f922d5ce03062302803ea5/tests/acceptance/Lab.robot).

If agree with the changes made for all your previous comments, I will rebase onto the `main` and add acceptance tests for `Notebook 7` as well.
